### PR TITLE
Add hot module reloading (fixes #199)

### LINKED ISSUE #199: Add hot module loading support

### PR DESCRIPTION
**Pull request by muffinresearch**
If you want to try it and it doesn't work, make sure you do have the latest grunt-webpack installed as per the package.json.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,6 +48,17 @@ are identical and can't get mangled by a broken deps installation.
 
 We're using webpack to build a JS bundle.
 
+### Hot module reloading
+
+If you run the webpack-dev-server you can get hot module reloading.
+
+For example run `grunt serve`
+
+And then visit http://localhost:8080/webpack-dev-server/management.html
```
Comment (kumar303) [inline README.md]: how hard is it to get it working on `pay.dev`? The main reason is that we need to be on the same domain as the API.

File: karma.shared.js [hunk 2]
```diff
@@ -2,16 +2,19 @@
 
 /* This is the shared karma config */
 
+var merge = require('lodash.merge');
 var RewirePlugin = require('rewire-webpack');
-var webpackConfig = require('./webpack.config.js');
+var webpackConfig = require('./webpack.config');
 
+
+var newWebpackConfig = merge({}, webpackConfig);
```
Comment (kumar303) [inline karma.shared.js]: this works but we also have the [object-assign](https://github.com/mozilla/payments-ui/blob/master/package.json#L16) polyfill which is used elsewhere like `assign({}, ...)`

**Comment by kumar303** [comment]
Nice! r+wc. I didn't try it out yet. It would be really great if we could do this on `pay.dev` (with a setting or whatever) but this is good start.

File: karma.shared.js (hunk 2 above)
Comment (kumar303) [inline karma.shared.js]: Thinking more about this, I'd prefer `object-assign` just so we don't add a new dep (even though it's a dev dep).

File: README.md (hunk 1 above)
Comment (muffinresearch) [inline README.md]: Hmm, yeah - will need to think about that. It'll probably need Charles Proxy or maybe we can do something with nginx?

File: karma.shared.js (hunk 2 above)
Comment (muffinresearch) [inline karma.shared.js]: Ah ok, I didn't know about that polyfill. I'll ditch the lodash dep.

Comment (muffinresearch) [inline karma.shared.js]: @kumar303 actually assign isn't going to work for this. See http://jsbin.com/nuhawi/edit?js,console for why.

File: README.md (hunk 1 above)
Comment (kumar303) [inline README.md]: As discussed, do the simplest thing first and land it :) Looking at the CORS cookie thing though, I think we can run the dev server on the host and talk to `pay.dev/api/` with just a few setting changes in the service.

File: karma.shared.js (hunk 2 above)
Comment (kumar303) [inline karma.shared.js]: Also as discussed, carry on! Maybe we could just remove assign everywhere for now? A quick grep suggests to me that we aren't even using it, at least not in prod code.

**Comment by kumar303** [comment]
ah, I added some comments thinking you hadn't merged this yet. You can ignore them. Glad this landed and is moving forward.

**Pull request merged**